chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -489,6 +621,193 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
+            .iter()
+            .filter_map(|x| match x.value() {
+                OutputValue::Coin(_) => None,
+                OutputValue::Token(asset) => match asset {
+                    TokenData::TokenTransferV1 { token_id, amount } => {
+                        (token_id == burn_token_id).then_some(amount)
+                    }
+                    TokenData::TokenIssuanceV1 {
+                        token_ticker: _,
+                        amount_to_issue: _,
+                        number_of_decimals: _,
+                        metadata_uri: _,
+                    }
+                    | TokenData::TokenBurnV1 {
+                        token_id: _,
+                        amount_to_burn: _,
+                    } => None,
+                },
+            })
+            .try_fold(Amount::from_atoms(0), |accum, output| accum + *output)
+            .ok_or(TokensError::CoinOrTokenOverflow)?;
+        Ok(change_amount)
+    }
+
+    fn check_connected_tokens(
+        &self,
+        token_data: &TokenData,
+        tx: &Transaction,
+        block: &Block,
+    ) -> Result<(), TokensError> {
+        match token_data {
+            TokenData::TokenTransferV1 { token_id, amount } => {
+                self.check_connected_transfer_data(block.get_id(), tx, token_id, amount)?;
+            }
+            TokenData::TokenIssuanceV1 {
+                token_ticker,
+                amount_to_issue,
+                number_of_decimals,
+                metadata_uri,
+            } => {
+                self.check_connected_issue_data(
+                    token_ticker,
+                    amount_to_issue,
+                    number_of_decimals,
+                    metadata_uri,
+                    tx.get_id(),
+                    block.get_id(),
+                )?;
+                self.register_new_token()?;
+            }
+            TokenData::TokenBurnV1 {
+                token_id,
+                amount_to_burn,
+            } => {
+                self.check_connected_burn_data(tx, &block.get_id(), token_id, amount_to_burn)?;
+            }
+        }
+        Ok(())
+    }
+
+    fn check_tokens_values(&self, block: &Block) -> Result<(), ConnectTransactionError> {
+        for tx in block.transactions() {
+            // Check tokens before connect tx
+            tx.outputs()
+                .iter()
+                .filter_map(|output| match output.value() {
+                    OutputValue::Coin(_) => None,
+                    OutputValue::Token(token_data) => Some(token_data),
+                })
+                .try_for_each(|token_data| self.check_connected_tokens(token_data, tx, block))
+                .map_err(ConnectTransactionError::TokensError)?;
+
+            // If it is not a genesis and in tx issuance
+            if self
+                .db_tx
+                .get_best_block_id()
+                .map_err(ConnectTransactionError::StorageError)?
+                .is_some()
```

Comment:
does this check make sense? I've understood that best block should always be available

Reply:
I think so. I guess, in the future, some tokens might be connected through genesis. If it is the genesis block, we won't check fees for issuance and issuance count. Correct me if I'm wrong @TheQuantumPhysicist  

However, I'm still working on it, and next commits make the code clearer.

Reply:
Fixed, this check was removed according to Sam's comments.